server/routes/referrals/referralDataPresenterUtils.test.ts

```diff
@@ -0,0 +1,252 @@
+          const utils = new ReferralDataPresenterUtils(referral, userInputData)
+
+          expect(utils.stringValue('additionalNeedsInformation', 'additional-needs-information')).toBe('100')
+        })
+      })
+    })
+  })
+
+  describe('booleanValue', () => {
+    describe('when the referral has a null value for the property', () => {
+      describe('when there is no user input data', () => {
+        it('returns null', () => {
+          const referral = draftReferralFactory.build({ needsInterpreter: null })
+          const utils = new ReferralDataPresenterUtils(referral, null)
+
+          expect(utils.booleanValue('needsInterpreter', 'needs-interpreter')).toBeNull()
+        })
+      })
+
+      describe('when there is user input data and the user input value for the property is null', () => {
+        it('returns null', () => {
+          const referral = draftReferralFactory.build({ needsInterpreter: null })
+          const userInputData = { 'needs-interpreter': null }
+          const utils = new ReferralDataPresenterUtils(referral, userInputData)
+
+          expect(utils.booleanValue('needsInterpreter', 'needs-interpreter')).toBeNull()
+        })
+      })
+
+      describe('when there is user input data and the property is absent from the user input data', () => {
+        it('returns null', () => {
+          const referral = draftReferralFactory.build({ needsInterpreter: null })
+          const userInputData = {}
+          const utils = new ReferralDataPresenterUtils(referral, userInputData)
+
+          expect(utils.booleanValue('needsInterpreter', 'needs-interpreter')).toBe(null)
+        })
+      })
+
+      describe("when there is user input data and the user input value for the property is 'no'", () => {
+        it('returns false', () => {
+          const referral = draftReferralFactory.build({ needsInterpreter: null })
+          const userInputData = { 'needs-interpreter': 'no' }
+          const utils = new ReferralDataPresenterUtils(referral, userInputData)
+
+          expect(utils.booleanValue('needsInterpreter', 'needs-interpreter')).toBe(false)
+        })
+      })
+
+      describe("when there is user input data and the user input value for the property is 'yes'", () => {
+        it('returns true', () => {
+          const referral = draftReferralFactory.build({ needsInterpreter: null })
+          const userInputData = { 'needs-interpreter': 'yes' }
+          const utils = new ReferralDataPresenterUtils(referral, userInputData)
+
+          expect(utils.booleanValue('needsInterpreter', 'needs-interpreter')).toBe(true)
+        })
+      })
+
+      describe("when there is user input data and the user input value for the property is present but neither 'yes', 'no' nor null", () => {
+        it('returns null', () => {
+          const referral = draftReferralFactory.build({ needsInterpreter: null })
+          const userInputData = { 'needs-interpreter': true }
```

Comment:
is this also the case for 'true' and 'false'?

Reply:
reason for asking is that i would naively expect 'truthy' strings to resolve properly. in the past i've written little helpers like `{ return ['true', 'on', 'yes', '1'].includes(val) }`.